File: backend/level3/Rental.py
from datetime import datetime
import Discount
import Commission
class Rental:
    def __init__(self,car_data,rental_data):
        #retrieving the dictionary in cars which has the same id as the car_id from rentals
        dictionary = list(filter(lambda car_data: car_data['id'] == rental_data["car_id"], car_data))[0]
        self.id = rental_data["id"]
        #finding the next set of values needed in the price formula
        self.price_per_km = dictionary["price_per_km"]
        self.price_per_day = dictionary["price_per_day"]
        self.distance = rental_data["distance"]
        self.numberofdays = Rental.count_days(rental_data["start_date"],rental_data["end_date"])
        #calculating the price
        self.discounted_rental = Discount.Rental_discount(self.numberofdays,self.price_per_day)
        self.price = int(self.discounted_rental.discount) + (self.distance* self.price_per_km)
        commission_rental = Commission.Rental_commission(self.numberofdays,self.price)
        self.commission = commission_rental.commission

    def count_days(start_date,end_date):
        date = "%Y-%m-%d"
        a = datetime.strptime(start_date, date)
        b = datetime.strptime(end_date, date)
        delta = b - a
        return delta.days + 1

File: backend/level3/Rental.rb
require_relative "Discount"
require_relative "Commission"
class Rental_price
    attr_reader :id, :price, :commission
    def initialize(car_data,rental_data) 
        #retrieving the dictionary in cars which has the same id as the car_id from rentals
        @dictionary = car_data.select {|k| k["id"] == rental_data["car_id"]}[0]
        @id = rental_data["id"]
        #finding the next set of values needed in the price formula
        @price_per_km = @dictionary["price_per_km"]
        @price_per_day = @dictionary["price_per_day"]
        @distance = rental_data["distance"]
        @numberofdays = Rental_price.days(rental_data["start_date"],rental_data["end_date"])
        #calculating the price
        @discounted_rental = Rental_discount.new(@numberofdays,@price_per_day)
        @price = @discounted_rental.discount + (@distance* @price_per_km)
        @commission_rental = Rental_commission.new(@numberofdays,@price)
        @commission = @commission_rental.commission
    end
    # creating a function to find the number of days between the start and end date
    def self.days(start_date,end_date)
        a = Date.parse(start_date)
        b = Date.parse(end_date)
        delta = b - a
        return delta.to_i + 1
    end 
end